lib/ansible/playbook/handler.py

```diff
@@ -25,10 +25,12 @@
 
 class Handler(Task):
 
-    _listen = FieldAttribute(isa='list')
+    _listen = FieldAttribute(isa='list', default=list)
 
     def __init__(self, block=None, role=None, task_include=None):
-        self._flagged_hosts = []
+        self.notified_hosts = []
+
+        self.cached_name = False
```

Comment:
I would use `None` since this is a string in the end and changing bool/string seems strange, while `None` is expected

Reply:
I am actually using it as a bool, since I was caching into the `.name` attr.  This is just a bool indicating whether or not it was cached.

I could however make this `None` and not overwrite `.name` and use `.cached_name` for the cache itself.

Do you think that is a better approach?

Reply:
mostly fine even as is, just minor point of possible future contention if we every really care about type enforcement